src/autowiring/demangle.cpp

```diff
@@ -7,28 +7,36 @@
 #include <cxxabi.h>
 #include <cstdlib>
 
-std::string autowiring::demangle(const std::type_info& ti) {
+std::string autowiring::demangle(const std::type_info& ti, bool* success) {
   int status;
   std::unique_ptr<char, void(*)(void*)> res{
     abi::__cxa_demangle(ti.name(), nullptr, nullptr, &status),
     std::free
   };
+  if(success != nullptr)
+    *success = status == 0;
+  
   return std::string(status == 0 ? res.get() : ti.name());
```

Comment:
Recommend we change the behavior of this routine to return an empty string when demangling is impossible, rather than extending the parameter space